app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx

```diff
@@ -180,6 +217,23 @@ export class CreatePipelineModal extends React.Component<ICreatePipelineModalPro
     return () => this.setState({useTemplate});
   }
 
+  private handleUseManagedTemplateSelection(useManagedTemplate: boolean): () => void {
+    return () => {
+      this.setState({
+        useManagedTemplate,
+        templateSourceUrl: '',
+        loadingTemplateFromSourceError: false,
+        command: Object.assign({}, this.state.command, {template: null}),
+      });
+    };
+  }
+
+  public handleSourceUrlChange(e: React.ChangeEvent<HTMLInputElement>): void {
+    const templateSourceUrl = e.target.value;
+    this.setState({templateSourceUrl});
+    this.loadPipelineTemplateFromSource(templateSourceUrl);
```

Comment:
Not sure if this _will_ cause any issues, but if the `loadPipelineTemplateFromSource` isn't debounced, you'll be calling setState back to back, which can sometimes have some odd rendering effects...